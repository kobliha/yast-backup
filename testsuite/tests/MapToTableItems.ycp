// Backup module - testsuite
//
// MapToTableItems function tests
//
// $Id$
//

{

include "backup/functions.ycp";
include "testsuite.ycp";

import "Backup";

// used global variables
string nocheckmark = "[ ]";
string checkmark = "[x]";

Backup::backup_files = nil;
Backup::selected_files = $["aaa_base-2001.5.22-0" : $["changed_files" : ["/etc/modules.conf"]]];

DUMP( MapToTableItems(nil) );

Backup::backup_files = $[];
DUMP( MapToTableItems(true) );

Backup::backup_files = $[
  "pkg1": $["changed_files" : []],
  "pkg2" : $["changed_files" : nil],
  "aaa_base-2001.5.22-0" : $["changed_files" : ["/etc/inittab", "/etc/modules.conf"]],
  "" : $["changed_files" : ["/etc/fstab", "/etc/passwd"]],
  "pkg3" : nil
];
DUMP( MapToTableItems(false) );

DUMP( MapToTableItems(nil) );

DUMP( MapToTableItems(true) );

}

