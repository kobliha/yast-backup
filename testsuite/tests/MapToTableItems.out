Read	.target.size "/etc/install.inf" -1
Read	.target.tmpdir nil
Read	.target.size nil -1
Dir	.autoinst.menuentry.s: []
Read	.probe.architecture nil
Read	.probe.has_apm nil
Read	.probe.has_pcmcia nil
Read	.probe.has_smp false
Read	.probe.system nil
Dump	[]
Dump	[]
Dump	[`item (`id (1), "[ ]", "/etc/fstab", ""), `item (`id (2), "[ ]", "/etc/passwd", ""), `item (`id (3), "[ ]", "/etc/inittab", "aaa_base-2001.5.22-0"), `item (`id (4), "[ ]", "/etc/modules.conf", "aaa_base-2001.5.22-0")]
Dump	[`item (`id (1), "[ ]", "/etc/fstab", ""), `item (`id (2), "[ ]", "/etc/passwd", ""), `item (`id (3), "[ ]", "/etc/inittab", "aaa_base-2001.5.22-0"), `item (`id (4), "[x]", "/etc/modules.conf", "aaa_base-2001.5.22-0")]
Dump	[`item (`id (1), "[x]", "/etc/fstab", ""), `item (`id (2), "[x]", "/etc/passwd", ""), `item (`id (3), "[x]", "/etc/inittab", "aaa_base-2001.5.22-0"), `item (`id (4), "[x]", "/etc/modules.conf", "aaa_base-2001.5.22-0")]
